### PR TITLE
Adopt AppStream

### PR DESCRIPTION
**Pull request by overflw**
With those small changes I hope to make tribler compatible with the [AppStream](https://www.freedesktop.org/software/appstream/docs/) standard. I couldn't test the changes yet, because tribler won't build for me at the moment.

### CODE REVIEW

**Comment by devos50** [comment]
ok to test

File: build/debian/tribler/usr/share/metainfo/org.tribler.Tribler.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      <image>https://www.tribler.org/img/image10.png</image>
+      <caption>Screenshot of the Tribler UI</caption>
+    </screenshot>
+  </screenshots>
+  <launchable type="desktop-id">org.tribler.Tribler.desktop</launchable>
+  <content_rating type="oars-1.1">
+    <content_attribute id="social-info">mild</content_attribute>
+  </content_rating>
+  <releases>
+    <release version="7.8.0" date="2021-02-15"/>
```
Comment (devos50) [inline build/debian/tribler/usr/share/metainfo/org.tribler.Tribler.metainfo.xml:47]: We refrain from hard-coding versions in our files. Please have a look at [this script](https://github.com/Tribler/tribler/blob/main/build/update_version_from_git.py) that is executed before packaging Tribler 👍

Comment (qstokkink) [reply build/debian/tribler/usr/share/metainfo/org.tribler.Tribler.metainfo.xml:47]: @overflw this is a way to satisfy this request for changes:

1. Remove the last three lines from this file (line 47, line 48 and line 49) and leave the last line blank.

2. Paste the following code into the `__main__` section of [that script](https://github.com/Tribler/tribler/blob/main/build/update_version_from_git.py):

```python
logger.info('Writing AppStream version.')
with open(path.join('build', 'debian', 'tribler', 'usr', 'share', 'metainfo',
                    'org.tribler.Tribler.metainfo.xml'), 'a') as f:
    f.write(f'    <release version="{version_id}" date="{build_date}"/>'
            f'\n  </releases>\n</component>')
```

Comment (overflw) [reply build/debian/tribler/usr/share/metainfo/org.tribler.Tribler.metainfo.xml:47]: Hey (:  thanks for the hints! I wrote a xml parsing based addendum to `update_version_from_git.py`:
```python
    logger.info('Writing AppStream version info.')
    releaseDate = time.strftime("%Y-%m-%d", time.localtime())
    attrib = {'version': f'{version_id}', 'date':f'{releaseDate}'}

    tree = xml.parse(path.join('build', 'debian', 'tribler', 'usr', 'share', 'metainfo',
                        'org.tribler.Tribler.metainfo.xml'))
    xmlRoot = tree.getroot()
    releases = xmlRoot.find('releases')
    release = xml.SubElement(releases, 'release', attrib)
    tree.write(path.join('build', 'debian', 'tribler', 'usr', 'share', 'metainfo',
                        'org.tribler.Tribler.metainfo.xml'))
```

**Review by devos50 — COMMENTED** [review]
Thanks for your changes! Could you look at/implement the suggestion raised by DeepCode? Also, please satisfy the flake8 check by making sure the import order is according to the standards 👍

**Review by devos50 — COMMENTED** [review]
Thanks for the changes! I just installed `defusedxml` on the machine that runs the flake8 checker. @xoriole do you think it is fine to add this as a build dependency or do you prefer to use the 'normal' XML parsing tools?

**Comment by overflw** [comment]
> Thanks for the changes! I just installed `defusedxml` on the machine that runs the flake8 checker. @xoriole do you think it is fine to add this as a build dependency or do you prefer to use the 'normal' XML parsing tools?

Tbh, defusdxml is not really necessary here, since we only parse our own metainfo.xml. But it won't hurt I guess.

**Comment by xoriole** [comment]
@devos50 @overflw  I'm fine with having it as a build dependency for Linux. I'd prefer it to be conditional import and the code for xml parsing and update to be under Linux section (Line 60).

**Comment by qstokkink** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Review by xoriole — DISMISSED** [review]

**Comment by overflw** [comment]
Hey, sorry for dismissing the review - I haven't seen it before pushing.. what can I do the close this pull request?

**Comment by devos50** [comment]
@overflw could you please squash your commit history into one commit (and remove the merge commits)?

**Comment by overflw** [comment]
> @overflw could you rebase and please squash your commit history into one commit (and remove the merge commits)?

Should be done, I hope.

**Comment by qstokkink** [comment]
Failure on Mac is due to #6042:

```
Error Message

assert 59852 == 59850   +59852   -59850

Stacktrace

def test_get_first_free_port():
        # target port is free
        port = random.randint(50000, 60000)
>       assert get_first_free_port(start=port) == port
E       assert 59852 == 59850
E         +59852
E         -59850

src/tribler-core/tribler_core/tests/test_network_utils.py:56: AssertionError
```

This is a random failure, so retesting this PR should fix it.

**Comment by qstokkink** [comment]
retest this please

**Comment by overflw** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Review by xoriole — APPROVED** [review]

**Pull request merged**